python_modules/dagster/dagster/core/asset_defs/assets_job.py

```diff
@@ -266,7 +266,8 @@ def build_deps(
                 input_def = assets_def.node_def.input_def_named(input_name)
                 if not input_def.dagster_type.is_nothing:
                     raise DagsterInvalidDefinitionError(
-                        f"Input asset '{asset_key.to_string()}' for asset '{asset_key}' is not "
+                        f"Input asset '{asset_key.to_string()}' for asset "
+                        f"'{next(iter(assets_def.asset_keys)).to_string()}' is not "
```

Comment:
should we maybe use the name of the node definition rather than the name of the first asset key? might be something that would be easier for the user to find

Reply:
I think that we should still use the name of the first asset key. In the case of source assets, we only have the name of the asset key to inform the user